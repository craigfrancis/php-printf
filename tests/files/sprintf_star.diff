     float(1.2345678901234567)
     float(1.2345678901234569E+100)
003- 1.2345678901
004- 1.23456789
005- 1.2345678901234569e+100
006- 1.2345678901234569E+100
007- 1.2345678901234569e+100
008- 1.2345678901234569E+100
009- foo
     
011-   1.234568
012-    1.23457
013-     foobar
004+ Warning: Undefined variable $args in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29
     
015-      1.235
016-       1.23
017-        foo
018- 
019- 1.2345678901
020- 1.2345678901
021- 1.2345678901
022-   1.234568
023-   1.234568
024-   1.234568
025-      1.235
026-      1.235
027-      1.235
028- 
029- Precision must be an integer
030- Precision must be between -1 and 2147483647
031- Precision -1 is only supported for %g, %G, %h and %H
032- Width must be greater than zero and less than 2147483647
006+ Fatal error: Uncaught TypeError: user_vprintf(): Argument #2 ($values) must be of type array, null given, called in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29 and defined in /Volumes/WebServer/Projects/php.fprint/functions.php:22
007+ Stack trace:
008+ #0 /Volumes/WebServer/Projects/php.fprint/functions.php(29): user_vprintf('%.*f\n', NULL)
009+ #1 /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_star.php(8): user_printf('%.*f\n', 10, 1.2345678901235)
010+ #2 {main}
011+   thrown in /Volumes/WebServer/Projects/php.fprint/functions.php on line 22
