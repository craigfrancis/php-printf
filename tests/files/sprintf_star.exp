float(1.2345678901234567)
float(1.2345678901234569E+100)
1.2345678901
1.23456789
1.2345678901234569e+100
1.2345678901234569E+100
1.2345678901234569e+100
1.2345678901234569E+100
foo

  1.234568
   1.23457
    foobar

     1.235
      1.23
       foo

1.2345678901
1.2345678901
1.2345678901
  1.234568
  1.234568
  1.234568
     1.235
     1.235
     1.235

Precision must be an integer
Precision must be between -1 and 2147483647
Precision -1 is only supported for %g, %G, %h and %H
Width must be greater than zero and less than 2147483647