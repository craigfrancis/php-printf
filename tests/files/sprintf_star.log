
---- EXPECTED OUTPUT
float(1.2345678901234567)
float(1.2345678901234569E+100)
1.2345678901
1.23456789
1.2345678901234569e+100
1.2345678901234569E+100
1.2345678901234569e+100
1.2345678901234569E+100
foo

  1.234568
   1.23457
    foobar

     1.235
      1.23
       foo

1.2345678901
1.2345678901
1.2345678901
  1.234568
  1.234568
  1.234568
     1.235
     1.235
     1.235

Precision must be an integer
Precision must be between -1 and 2147483647
Precision -1 is only supported for %g, %G, %h and %H
Width must be greater than zero and less than 2147483647
---- ACTUAL OUTPUT
float(1.2345678901234567)
float(1.2345678901234569E+100)

Warning: Undefined variable $args in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29

Fatal error: Uncaught TypeError: user_vprintf(): Argument #2 ($values) must be of type array, null given, called in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29 and defined in /Volumes/WebServer/Projects/php.fprint/functions.php:22
Stack trace:
#0 /Volumes/WebServer/Projects/php.fprint/functions.php(29): user_vprintf('%.*f\n', NULL)
#1 /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_star.php(8): user_printf('%.*f\n', 10, 1.2345678901235)
#2 {main}
  thrown in /Volumes/WebServer/Projects/php.fprint/functions.php on line 22
---- FAILED
