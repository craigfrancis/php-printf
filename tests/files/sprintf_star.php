<?php


$f = 1.23456789012345678;
$fx = 1.23456789012345678e100;
var_dump($f, $fx);

user_printf("%.*f\n", 10, $f);
user_printf("%.*G\n", 10, $f);
user_printf("%.*g\n", -1, $fx);
user_printf("%.*G\n", -1, $fx);
user_printf("%.*h\n", -1, $fx);
user_printf("%.*H\n", -1, $fx);
user_printf("%.*s\n", 3, "foobar");
echo "\n";

user_printf("%*f\n", 10, $f);
user_printf("%*G\n", 10, $f);
user_printf("%*s\n", 10, "foobar");
echo "\n";

user_printf("%*.*f\n", 10, 3, $f);
user_printf("%*.*G\n", 10, 3, $f);
user_printf("%*.*s\n", 10, 3, "foobar");
echo "\n";

user_printf("%1$.*2\$f\n", $f, 10);
user_printf("%.*2\$f\n", $f, 10);
user_printf("%2$.*f\n", 10, $f);
user_printf("%1$*2\$f\n", $f, 10);
user_printf("%*2\$f\n", $f, 10);
user_printf("%2$*f\n", 10, $f);
user_printf("%1$*2$.*3\$f\n", $f, 10, 3);
user_printf("%*2$.*3\$f\n", $f, 10, 3);
user_printf("%3$*.*f\n", 10, 3, $f);
echo "\n";

try {
    user_printf("%.*G\n", "foo", 1.5);
} catch (ValueError $e) {
    echo $e->getMessage(), "\n";
}

try {
    user_printf("%.*G\n", -100, 1.5);
} catch (ValueError $e) {
    echo $e->getMessage(), "\n";
}

try {
    user_printf("%.*s\n", -1, "Foo");
} catch (ValueError $e) {
    echo $e->getMessage(), "\n";
}

try {
    user_printf("%*G\n", -1, $f);
} catch (ValueError $e) {
    echo $e->getMessage(), "\n";
}

?>
