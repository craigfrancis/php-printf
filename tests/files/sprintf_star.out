float(1.2345678901234567)
float(1.2345678901234569E+100)

Warning: Undefined variable $args in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29

Fatal error: Uncaught TypeError: user_vprintf(): Argument #2 ($values) must be of type array, null given, called in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29 and defined in /Volumes/WebServer/Projects/php.fprint/functions.php:22
Stack trace:
#0 /Volumes/WebServer/Projects/php.fprint/functions.php(29): user_vprintf('%.*f\n', NULL)
#1 /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_star.php(8): user_printf('%.*f\n', 10, 1.2345678901235)
#2 {main}
  thrown in /Volumes/WebServer/Projects/php.fprint/functions.php on line 22